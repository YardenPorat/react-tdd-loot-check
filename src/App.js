import React from 'react';
import Wallet from './components/Wallet';
import Loot from './components/Loot';

const App = () => {
  return (
    <div className='App'>
      <h2>Loot Check</h2>
      <hr />
      <Wallet />
      <hr />
      <Loot />
      <div>
        Rates by{' '}
        <a
          target='_blank'
          rel='noreferrer'
          href='https://www.coindesk.com/price'
        >
          Coindesk
        </a>
      </div>
    </div>
  );
};

export default App;
