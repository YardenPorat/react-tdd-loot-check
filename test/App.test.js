import React from 'react';
import App from '../src/App';
import { shallow } from 'enzyme';
import { expect } from 'chai';

describe('renders learn react link', () => {
  const app = shallow(<App />);
  it('renders properly', () => {
    expect(app).to.matchSnapshot(true);
  });

  it('contains a Connect(Wallet) component', () => {
    expect(app.find('Connect(Wallet)').exists()).to.eq(true);
  });

  it('contains a Connect(Loot) component', () => {
    expect(app.find('Connect(Loot)').exists()).to.eq(true);
  });

  it('contains a link to the coindesk price page', () => {
    expect(app.find('a').props().href).to.eq('https://www.coindesk.com/price');
  });
});
