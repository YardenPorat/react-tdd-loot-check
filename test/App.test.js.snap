// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Loot Check
        </h2>,
        <hr />,
        <UNDEFINED />,
        <hr />,
        <UNDEFINED />,
        <div>
          Rates by
           
          <a
            href="https://www.coindesk.com/price"
            rel="noreferrer"
            target="_blank"
          >
            Coindesk
          </a>
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loot Check",
        },
        "ref": null,
        "rendered": "Loot Check",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Rates by",
            " ",
            <a
              href="https://www.coindesk.com/price"
              rel="noreferrer"
              target="_blank"
            >
              Coindesk
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Rates by",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Coindesk",
              "href": "https://www.coindesk.com/price",
              "rel": "noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Coindesk",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Loot Check
          </h2>,
          <hr />,
          <UNDEFINED />,
          <hr />,
          <UNDEFINED />,
          <div>
            Rates by
             
            <a
              href="https://www.coindesk.com/price"
              rel="noreferrer"
              target="_blank"
            >
              Coindesk
            </a>
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loot Check",
          },
          "ref": null,
          "rendered": "Loot Check",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Rates by",
              " ",
              <a
                href="https://www.coindesk.com/price"
                rel="noreferrer"
                target="_blank"
              >
                Coindesk
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Rates by",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coindesk",
                "href": "https://www.coindesk.com/price",
                "rel": "noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Coindesk",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
